tests/bwc/test_upgrade.py

```diff
@@ -569,3 +570,62 @@ def test_snapshot_compatibility(self):
                 run_selects(c, self.VERSION[-1])
             shutil.rmtree(path_data, ignore_errors=True)
         self._process_on_stop()
+
+
+class ReturningUpdateNodesCompatibilityTest(NodeProvider, unittest.TestCase):
+
+    VERSION = ('4.1', 'latest-nightly')
+
+    CLUSTER_SETTINGS = {
+        'cluster.name': gen_id(),
+    }
+
+    def test_update_returning_in_mixed_cluster(self):
+
+        """Test backward compatibility when using update with returning clause
+           in a mixed cluster of 4.1 and 4.2 nodes. This test ensures that the
+           previous functionality from 4.1 for update is fully working when a
+           4.2 node is in the cluster. It also ensures that the use of a returning
+           clause from a 4.2 node yields a meaningful error when used in a cluster
+           with 4.1 nodes.
+        """
+
+        cluster = self._new_heterogeneous_cluster(self.VERSION, random.randint(3, 9), self.CLUSTER_SETTINGS)
+        cluster.start()
+
+        node_4_1 = next(x for x in cluster._nodes if '4.1' in x.crate_dir)
+        node_4_2 = next(x for x in cluster._nodes if '4.2' in x.crate_dir)
```

Comment:
It think it would be nice if the version would stored directly on the `CrateNode` so i do have scan the `crate_dir`.

Reply:
I think currently there isn't.

I agree, it would probably be an option to expose `version` in `CrateNode`